packages/components/src/autocomplete/types.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * WordPress dependencies
+ */
+import type { WPElement } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import type { useAutocomplete } from '.';
+
+type GetOptions< TCompleterOption = any > = (
+	filterValue: AutocompleterUIProps[ 'filterValue' ]
+) => Array< TCompleterOption > | Promise< Array< TCompleterOption > >;
+
+export type OptionCompletion = {
+	action: 'insert-at-caret' | 'replace';
```

Comment:
Borrowing from definitely typed, would split `Action` to its separate type and adding these docs help?

Reply:
With the current implementation I think adding the docs might be helpful, but the separate `Action` type doesn't look like it would be used, unless I'm overlooking something.

That said, this would be impacted by the potential change I've shared [here](https://github.com/WordPress/gutenberg/pull/47751#discussion_r1108744065), because at that point `action` is even more narrowly defined, depending on which `value` type it's being paired with. In that context, would you still like to see some docs here?

Reply:
We may add the docs directly to the `OptionCompletion` type or to the `getOptionCompletion` function?

Reply:
Marking as solved, as this will likely be part of the changes proposed in https://github.com/WordPress/gutenberg/pull/47751#discussion_r1112760535